packages/@aws-cdk/aws-servicecatalogappregistry/lib/aspects/stack-associator.ts

```diff
@@ -89,33 +107,43 @@ abstract class StackAssociatorBase implements IAspect {
     }
 
     if (node.account != this.application.env.account && !this.sharedAccounts.has(node.account)) {
-      this.application.shareApplication({
-        accounts: [node.account],
-        sharePermission: SharePermission.ALLOW_ACCESS,
-      });
+      if (this.enableCrossAccount) {
+        this.application.shareApplication({
+          accounts: [node.account],
+          sharePermission: SharePermission.ALLOW_ACCESS,
+        });
 
-      this.sharedAccounts.add(node.account);
+        this.sharedAccounts.add(node.account);
+      } else {
+        this.warning(node, 'Cross-account stack detected but application sharing and association will be skipped because cross-account option is not enabled.');
```

Comment:
why is this not an error? It sounds like the user wanted to enable cross-stack, but that they didn't enable it. That's a bad configuration => it should throw an error

Reply:
It is a warning instead of an error to not block customers from proceeding with the deployment of cross-account stacks. In the CDK app definition, the user should be able to specify stacks owned by other accounts, and as long as the accounts that own those stacks have applied the permissions to allow CDK deployments, then they should be allowed.

The warning states that the stack would not be associated with the AppRegistry Application, but otherwise, the deployment of the resources can proceed. Later, if the customer wishes, when the account of the stack is configured to be part of the same AWS Organization as the Application owner, and cross-account sharing is enabled, only then can the customer enable the option to perform the cross-account association.